scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -2599,6 +2599,13 @@ class Router(formatOps: FormatOps) {
       case ft @ FormatToken(_: T.Comment, _, _) if ft.hasBreak =>
         splitsAsNewlines(splits)
       case ft if ft.meta.formatOff && ft.hasBreak => splitsAsNewlines(splits)
+      case FormatToken(_: T.Equals, r: T.KwMacro, _)
+          if dialect.allowInlineMods =>
```

Comment:
@tgodzik not sure if this is the right way to identify scala 3, perhaps as good as any

Reply:
I would use `allowSignificantIndentation`, because it's the cause of the issue. Without significant indentation there would be no issues